### PR TITLE
TTML to VTT style migration

### PR DESCRIPTION
**Pull request by Avishek-Gulshan**
- Update TTM L parser to push parent style to child tags
- Creating new custom style classes by combining inline style and style class
- Added additional style propagation from TTML to VTT

### CODE REVIEW

**Comment by Avishek-Gulshan** [comment]
Hi,

Can you please have a look at this PR. See if this makes sense

**Comment by Avishek-Gulshan** [comment]
Hi, Please take a look at this PR

**Comment by asticode** [comment]
This is a big PR and I'm pretty busy right now, I'll need some time to review it

File: ttml.go [hunk 1]
```diff
@@ -422,13 +507,186 @@ func ReadFromTTML(i io.Reader) (o *Subtitles, err error) {
...
+			line.Items = lineItems
+			lines = append(lines, line)
+		}
+		item.Lines = lines
+	}
+
 	return
 }
 
+func checkIfCustomStyleExists(styleMap map[Style]string, style *Style) (bool, string) {
```
Comment (asticode) [inline ttml.go:601]: Could you transform this to `func (ss ttmlCustomStyles) exists(s *Style) (id string, ok bool)` instead ?

File: ttml.go [hunk 2]
```diff
@@ -422,13 +507,186 @@ func ReadFromTTML(i io.Reader) (o *Subtitles, err error) {
...
+						inlineStyle = combineInlineStyles(lineItem.Style.InlineStyle, *lineItem.InlineStyle)
+					}
+					newStyle := &Style{
+						InlineStyle: inlineStyle,
+					}
+					if lineItem.Style != nil {
+						newStyle.Style = lineItem.Style.Style
+					}
+					newStyle.InlineStyle.propagateTTMLAttributes()
+					ok, id := checkIfCustomStyleExists(customStyleMap, newStyle)
```
Comment (asticode) [inline ttml.go:578]: Could you make this a one liner : `if ok, id := checkIfCustomStyleExists(customStyleMap, newStyle); ok {`

File: ttml.go [hunk 3]
```diff
@@ -422,13 +507,186 @@ func ReadFromTTML(i io.Reader) (o *Subtitles, err error) {
...
+			ts.TTMLInStyleAttributes.WritingMode = finalWritingMode
+		}
+		s.InlineStyle = ts.TTMLInStyleAttributes.styleAttributes()
 		// Append subtitle
 		o.Items = append(o.Items, s)
 	}
+
+	//Pushing Inline style from parent tag to every child tag then combining it with child tag's Style to create a new style and then assigning that style to the child tag.
+	customStyleNumber := 1
+	var customStyleMap = make(map[Style]string)
```
Comment (asticode) [inline ttml.go:536]: I feel it should be the other way around and with a pointer `var customStyleMap = make(map[string]*Style)`, and you should rename this variables to `customStyles`

**Review by asticode — CHANGES_REQUESTED** [review]
This is a huge PR, I've not reviewed everything but let start from here.

You need to fix tests as well.

**Comment by NhanNguyen700** [comment]
This one looks interesting, any further work?

**Comment by asticode** [comment]
Requested changes need to be addressed (this was the first part of my review) but based on the fact that this PR is pretty old, I doubt the OP will ever do it.

If you feel like it, you can pick it up 👍

**Pull request closed without merging**